src/YieldDisburser.sol

```diff
@@ -247,33 +239,26 @@ contract YieldDisburser is OwnableUpgradeable {
 
         BREAD.claimYield(BREAD.yieldAccrued(), address(this));
         uint256 projectsLength = projects.length;
-        (uint256[] memory projectDistributions, uint256 totalVotes) = _commitVotedDistribution();
-        if (totalVotes == 0) {
-            projectDistributions = new uint256[](projectsLength);
-            for (uint256 i; i < projectsLength; ++i) {
-                projectDistributions[i] = 1;
-            }
-            totalVotes = projectsLength;
-        }
 
         lastClaimedTimestamp = Time.timestamp();
         lastClaimedBlockNumber = Time.blockNumber();
-        currentVotes = 0;
-
         uint256 halfBalance = BREAD.balanceOf(address(this)) / 2;
         uint256 baseSplit = halfBalance / projectsLength;
         uint256 percentageOfTotalVote;
         uint256 votedSplit;
         uint256[] memory votedSplits = new uint256[](projectsLength);
         uint256[] memory percentages = new uint256[](projectsLength);
         for (uint256 i; i < projectsLength; ++i) {
-            percentageOfTotalVote = projectDistributions[i] / totalVotes;
-            votedSplit = halfBalance * (projectDistributions[i] * PRECISION / totalVotes) / PRECISION;
+            percentageOfTotalVote = projectDistributions[i] / currentVotes;
+            votedSplit = halfBalance * (projectDistributions[i] * PRECISION / currentVotes) / PRECISION;
             BREAD.transfer(projects[i], votedSplit + baseSplit);
             votedSplits[i] = votedSplit;
             percentages[i] = percentageOfTotalVote;
         }
         _updateBreadchainProjects();
+        delete currentVotes;
+        delete projectDistributions;
+        projectDistributions = new uint256[](projects.length);
         emit YieldDistributed(votedSplits, baseSplit, percentages, projects);
```

Comment:
imo there is no point in calculating the percentages o chain. just emit the total votes here and use `votedSplits` and `currentVotes` (which ideally is renamed to "totalCycleVotes" or something to be more clear) to calculate the values off chain on the frontend.

Reply:
@subject026 did you already integrate this? we can patch this later but I don't want to invoke more frontend changes atm

Reply:
No these events aren't being consumed anywhere yet so feel free to change them.